src/util/route.js

```diff
@@ -96,8 +97,8 @@ export function isSameRoute (a: Route, b: ?Route): boolean {
 function isObjectEqual (a = {}, b = {}): boolean {
   // handle null value #1566
   if (!a || !b) return a === b
-  const aKeys = Object.keys(a).sort()
-  const bKeys = Object.keys(b).sort()
+  const aKeys = Object.keys(a)
+  const bKeys = Object.keys(b)
```

Comment:
Why should this be removed?

Reply:
if that not be removed, they are the same object as below:
```
var a = {a:1,b:2}; var b = {b:2,a:1};

```

Object.keys method will keep the order of attributes in an object as the attributes  was added.

if that not be removed,  "https://vesaas.com/?a=1&b=2" and  "https://vesaas.com/?b=2&a=1" will be thought the same route;
```
  ensureURL (push?: boolean) {
    const location = getHash()
    const plainRoute = createPlainRoute(location)
    if (!isSameRoute(plainRoute, this.current)) {
      const current = this.current.fullPath
      push ? pushHash(current) : replaceHash(current)
    }
  }
``` 
 the "ensureURL " method will not go into 'if ' block in this scene.